server.go

```diff
@@ -0,0 +1,354 @@
+package r2go
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/opds-community/libopds2-go/opds2"
+	"github.com/readium/r2-streamer-go/pkg/decoder/lcp"
+	"github.com/readium/r2-streamer-go/pkg/fetcher"
+	"github.com/readium/r2-streamer-go/pkg/parser"
+	"github.com/readium/r2-streamer-go/pkg/pub"
+	"github.com/readium/r2-streamer-go/pkg/searcher"
+	"github.com/urfave/negroni"
+)
+
+type R2GoServer struct {
```

Comment:
Maybe `PublicationServer` or `PubServer` instead of `R2Go`?

```suggestion
type PublicationServer struct {
```